Improved performance of "all files" setting

robotboy655 opened the pull request:
Use `glualinter`'s ability to lint multiple files at once to severely cut down on number of processes spawned.

For GMod's base Lua code, the number goes from 619 to around 60, which significantly improves performance on Windows.

Related discussion: https://github.com/FPtje/GLuaFixer/issues/175

robotboy655 committed:
Improved performance of "all files" setting

Use glualinter's ability to lint multiple files at once to severely cut down on number of processes spawned.

For GMod's base Lua code, the number goes from 619 to around 60.

Goz3rr commented on src/LintProcess.ts:
```diff
@@ -3,12 +3,17 @@ import { spawn, ChildProcess } from 'child_process';
-    public StdOut: string = '';
-    public StdErr: string = '';
+    private StdOut: string = '';
+    //private StdErr: string = '';
```
Just delete the code instead of commenting it out if its not needed anymore, it'll be in the git history anyways

Goz3rr commented on src/glualintProvider.ts:
```diff
@@ -32,6 +33,28 @@ export default class GLuaLintingProvider implements vscode.Disposable {
+        const allFiles = config.get<boolean>('allFiles');
+        if (allFiles) {
+            // HACK: bypasses activeLanguages setting!
+            const allFiles = await vscode.workspace.findFiles('**/*.lua');
```
Something I've only now realized is that this might not work with virtual workspaces. I think I already noted somewhere that only single file linting works in this case, but might want to make sure the error checking below does not blow up incase you attempt to do it anyways.

robotboy655 replied:
It didn't blow up with virtual workspaces in my testing, but I went ahead and handled it better for this use case, so it will now fallback to the previous method of linting all files separately and actually work as a result.

I have also applied a concurrency limit to `lintUri` to ensure we only spawn 50 processes a time so the PC is not too swamped for large projects.

Goz3rr commented on src/glualintProvider.ts:
```diff
@@ -125,11 +141,28 @@ export default class GLuaLintingProvider implements vscode.Disposable {
+
+        const lintProcess: LintProcess = new LintProcess(uris[0], args);
+        if (!lintProcess.isValid()) {
+            // TODO: User friendly warning?
```
Probably use vscode.window.showErrorMessage here

Goz3rr reviewed (commented):
Cool, thanks again.

Few minor points that I noticed, once those are solved I'll merge and figure out how to make a release again, since it's been a while.

robotboy655 committed:
Requested changes

robotboy655 committed:
Handle virtual workspaces better

robotboy655 committed:
Apply a concurrency limit to lintUri

This ensures we only spawn 50 processes at a time when linting the whole workspace file by file, for virtual workspaces

Goz3rr commented:
Thanks 👍
